server/channels/app/ai.go

```diff
@@ -0,0 +1,143 @@
+
+Example format: "Team decided to migrate to microservices architecture [PERMALINK:%s/%s/pl/abc123xyz]"
+
+Your response must be compacted valid JSON only, with no additional text, formatting, nor code blocks.`, channelName, siteURL, teamName, conversationText, strings.Join(postIDs, ", "), siteURL, teamName, siteURL, teamName)
+
+	// Create bridge client
+	sessionUserID := ""
+	if session := rctx.Session(); session != nil {
+		sessionUserID = session.UserId
+	}
+	client := a.getBridgeClient(sessionUserID)
+
+	completionRequest := agentclient.CompletionRequest{
+		Posts: []agentclient.Post{
+			{Role: "system", Message: systemPrompt},
+			{Role: "user", Message: userPrompt},
+		},
+	}
+
+	rctx.Logger().Debug("Calling AI agent for post summarization",
+		mlog.String("channel_name", channelName),
+		mlog.String("user_id", userID),
+		mlog.String("agent_id", agentID),
+		mlog.Int("post_count", len(posts)),
+	)
+
+	completion, err := client.AgentCompletion(agentID, completionRequest)
+	if err != nil {
+		rctx.Logger().Error("AI agent call failed for summarization",
+			mlog.Err(err),
+			mlog.String("channel_name", channelName),
+			mlog.String("agent_id", agentID),
+		)
+		return nil, model.NewAppError("SummarizePosts", "app.ai.summarize.agent_call_failed", nil, err.Error(), 500)
+	}
+
+	var summary model.AIRecapSummaryResponse
+	if err := json.Unmarshal([]byte(completion), &summary); err != nil {
+		rctx.Logger().Error("Failed to parse AI summarization response",
+			mlog.Err(err),
+			mlog.String("response", completion),
+		)
+		return nil, model.NewAppError("SummarizePosts", "app.ai.summarize.parse_failed", nil, err.Error(), 500)
+	}
+
+	// Ensure arrays are never nil
+	if summary.Highlights == nil {
+		summary.Highlights = []string{}
+	}
+	if summary.ActionItems == nil {
+		summary.ActionItems = []string{}
+	}
+
+	rctx.Logger().Debug("AI summarization successful",
+		mlog.String("channel_name", channelName),
+		mlog.Int("highlights_count", len(summary.Highlights)),
+		mlog.Int("action_items_count", len(summary.ActionItems)),
+	)
+
+	return &summary, nil
+}
+
+func buildConversationText(posts []*model.Post) string {
+	text, _ := buildConversationTextWithIDs(posts)
+	return text
+}
+
+func buildConversationTextWithIDs(posts []*model.Post) (string, []string) {
+	var sb strings.Builder
+	postIDs := make([]string, 0, len(posts))
+
+	for _, post := range posts {
+		// Collect post ID
+		postIDs = append(postIDs, post.Id)
+
+		// Posts should have Username populated by the caller
+		// For posts without username, use UserId as fallback
+		username := ""
+		if usernameProp := post.GetProp("username"); usernameProp != nil {
+			if usernameStr, ok := usernameProp.(string); ok {
+				username = usernameStr
+			}
+		}
+		if username == "" {
+			username = post.UserId
+		}
+		sb.WriteString(fmt.Sprintf("[%s] %s (Post ID: %s): %s\n",
```

Comment:
Should this be translatable?

Reply:
The Agents plugin currently only officially supports english. So think for now, no.